backend/roles/team-roles-manager.js

```diff
@@ -2,45 +2,62 @@
 
 const twitchApi = require("../twitch-api/api");
 const frontendCommunicator = require("../common/frontend-communicator");
+const accountAccess = require("../common/account-access");
 
 /**
- * @param {import('twitch/lib/API/Kraken/Team/Team').Team[]} teams
+ * @typedef MappedTeamRole
+ * @property {string} id
+ * @property {string} name
  */
-function mapRoles(teams) {
+
+/** @returns {MappedTeamRole} */
+const mapRoles = (teams) => {
     return teams
         .map(team => {
             return {
-                id: team.id,
+                id: parseInt(team.id), // For compatibility (Kraken API returned an int)
                 name: team.displayName
             };
         });
-}
-
-async function getAllTeamRolesForViewer(username) {
-    const roles = await twitchApi.teams.getMatchingTeamsByName(username);
+};
 
-    return mapRoles(roles);
-}
+/** @returns {Promise.<MappedTeamRole[]>} */
+const getAllTeamRolesForViewer = async (username) => {
+    const client = twitchApi.getClient();
+    const user = await client.helix.users.getUserByName(username);
+    const streamerTeams = await client.helix.teams.getTeamsForBroadcaster(accountAccess.getAccounts().streamer.userId);
 
-async function getTeamRoles() {
-    const teams = await twitchApi.teams.getStreamerTeams();
+    if (streamerTeams == null) return null;
 
-    if (teams == null) {
-        return null;
+    const teams = [];
+    for (const team of streamerTeams) {
+        const relations = await team.getUserRelations();
+        for (const relation of relations) {
+            if (relation.id === user.id) {
+                teams.push(team);
+            }
+        }
```

Comment:
The outer `for` loop is unfortunately necessary because of the `await` call, but we could remove this nested one by instead doing:
```js
if(relations.some(r => r.id === user.id)) {
   teams.push(team);
}
```

Reply:
Fixed.